src/services/activityReports.js

```diff
@@ -21,12 +21,50 @@ import {
   NextStep,
   Objective,
   Program,
+  ActivityReportObjective,
 } from '../models';
 
 import { saveGoalsForReport } from './goals';
 
 import { saveObjectivesForReport } from './objectives';
 
+export async function batchQuery(query, limit) {
+  let finished = false;
+  let page = 0;
+  const finalResult = [];
+
+  while (!finished) {
+    // eslint-disable-next-line no-await-in-loop
+    const rows = await ActivityReport.findAll({
+      ...query,
+      limit,
+      offset: page * limit,
+    });
+    /*
+      Sequelize adds a bunch of data/functions to items it retrieves from
+      the database. We _should_ be able to give sequelize `raw: true` to
+      get results without the extra sequelize "stuff", but the link to an
+      issue below shows sequelize can't handle `raw: true` with `hasMany`
+      associations.
+
+      When DB objects have the extra sequelize "stuff" we run into memory
+      issues. When we get only data from the DB we don't run into memory
+      issues because all the sequelize "stuff" we don't need is garbage
+      collected at some point.
+
+      See https://github.com/sequelize/sequelize/issues/3897
+    */
+    const raw = JSON.parse(JSON.stringify(rows));
+    finalResult.push(...raw);
```

Comment:
not saying that there is a better way, but this is kind of an expensive operation and does not reflect well on sequelize for making us do it

Reply:
Yeah definitely. Especially because the issue in the comments was created in 2015